binding/org.eclipse.smarthome.binding.yahooweather/src/main/java/org/eclipse/smarthome/binding/yahooweather/handler/YahooWeatherHandler.java

```diff
@@ -64,8 +64,13 @@ public void initialize() {
         Configuration config = getThing().getConfiguration();
 
         location = (String) config.get("location");
-        if ("us".equalsIgnoreCase((String) config.get("unit"))) {
+        if ("metric".equalsIgnoreCase((String) config.get("unit"))) {
+            unit = "c";
+        } else if ("us".equalsIgnoreCase((String) config.get("unit"))) {
             unit = "f";
+        } else {
+            // means unit not known
+            unit = null;
```

Comment:
OK, you are right. But in general it means, that it should set the thing into the status "MISCONFIGURED". So we can change the binding, after we introduced the new status.